src/__tests__/FusionAuthProvider.test.tsx

```diff
@@ -64,12 +59,50 @@ describe('FusionAuthProvider', () => {
             wrapper,
         });
 
-        await waitFor(() =>
-            expect(result.current.user).toEqual({ name: 'trent anderson' }),
-        );
+        expect(result.current.user).toEqual(trent);
+        expect(result.current.isAuthenticated).toBe(true);
+    });
+
+    test('Will fetch the user from the server when the id token cookie is set', async () => {
+        Object.defineProperty(document, 'cookie', {
+            writable: true,
+            value: `app.idt=12345`,
+        });
+
+        const user = { name: 'Mr. Userton' };
+        mockFetchJson(user);
+
+        const serverUrl = 'my-server.com';
+        const mePath = '/my-me-path';
+
+        const { result } = renderHook(() => useFusionAuth(), {
+            wrapper: ({ children }) => (
+                <FusionAuthProvider
+                    {...TEST_CONFIG}
+                    serverUrl={serverUrl}
+                    mePath={mePath}
+                >
+                    {children}
+                </FusionAuthProvider>
+            ),
+        });
+
+        expect(result.current.isLoading).toBe(true);
+        expect(result.current.user).toEqual({});
+        expect(result.current.isAuthenticated).toBe(false);
+        expect(fetch).toHaveBeenCalledWith(serverUrl + mePath, {
+            credentials: 'include',
+        });
+
+        await waitFor(() => {
+            expect(result.current.isLoading).toBe(false);
+            expect(result.current.user).toEqual(user);
+            expect(result.current.isAuthenticated).toBe(true);
+            expect(fetch).toHaveBeenCalledTimes(1);
+        });
     });
 
-    test('User to empty when user cookie is not json parsable', async () => {
+    test('User to empty when user cookie is not json parsable', () => {
```

Comment:
@JakeLo123 

This test name is a little funky, are we trying to test that the "the user object should be empty when user cookie is not valid"?

Reply:
I didn't write that description, so I can't say for sure, but I would interpret it that way... "json parse for user fails `user` is an empty object"